chain/bitcoind_conn.go

```diff
@@ -437,6 +437,70 @@ func (c *BitcoindConn) GetBlock(hash *chainhash.Hash) (*wire.MsgBlock, error) {
 	}
 }
 
+// GetBlocksBatch returns a batch of raw blocks from the server given their
+// hashes. If the server has already pruned some of the blocks, it will be
+// retrieved from their peers.
+func (c *BitcoindConn) GetBlocksBatch(
+	hashes []*chainhash.Hash) ([]*wire.MsgBlock, error) {
+
+	batchRequest := make([]rpcclient.FutureGetBlockResult, 0, len(hashes))
+
+	for i := range hashes {
+		sendRequest := c.batchClient.GetBlockAsync(hashes[i])
+		batchRequest = append(batchRequest, sendRequest)
+	}
+
+	err := c.batchClient.Send()
+	if err != nil {
+		return nil, err
+	}
+
+	batchRawBlocks := make([]*wire.MsgBlock, 0, len(hashes))
+
+	for i := range hashes {
+		block, err := batchRequest[i].Receive()
+		// Got the block from the backend successfully, add to the response.
+		if err == nil {
+			batchRawBlocks = append(batchRawBlocks, block)
+			continue
+		}
+
+		// We failed getting the block from the backend for whatever
+		// reason. If it wasn't due to the block being pruned, return
+		// the error immediately.
+		if !isBlockPrunedErr(err) || c.prunedBlockDispatcher == nil {
+			return nil, err
+		}
+
+		// Now that we know the block has been pruned for sure, request
+		// it from our backend peers.
+		blockChan, errChan := c.prunedBlockDispatcher.Query(
+			[]*chainhash.Hash{hashes[i]},
+		)
+	out:
```

Comment:
Could extract the whole code of this loop into a function so we could use `return` instead of `continue` and `break out` which would increase readability.